starlite/testing.py

```diff
@@ -368,8 +384,14 @@ def _create_scope(
             user=user,
             auth=auth,
             query_string=urlencode(query_params, doseq=True).encode() if query_params else b"",
-            path_params=[],
+            path_params=path_params,
             client=(self.server, self.port),
+            state=state,
+            asgi=ASGIVersion(spec_version="3.0", version="3.0"),
+            http_version=http_version or "1.1",
+            raw_path=path.encode("ascii"),
+            route_handler=None,  # type: ignore[typeddict-item]
```

Comment:
I'm not sure what is the right way to let users choose the `route_handler`. As per my understanding, there is no easy way to extract the handler from the `app` object, and letting the users choose just any handler is probably not useful. Any idea how to implement that?

Reply:
Is route handler required to construct scope?

Reply:
It's a non-optional parameter in [`BaseScope`](https://github.com/starlite-api/starlite/blob/d6313981c9129da2f88efb09a9d14424733c781b/starlite/types/asgi_types.py#L78) but not required to create the `Request` object

Reply:
Can you use `app.route_handler_method_map().keys()`. This will give you all the registered route handler instances.

Reply:
But what if the app starts with `route_handlers=[]` or if it starts with more than one `route_handlers`? Which one should we choose?

Reply:
i wouldnt auto pick a value - i would suggest the following:

1. create a default placeholder, and use it if the user doesnt pass anything.
2. allow the user to pass a handler in manually. Consider this:

```python
@get()
def handler() -> None:
   ...

request = RequestFactory().get(path="/", route_handler=handler)
```

Reply:
Using raw_path and method, you can resolve its associated route handler by using `app.route_handler_method_map`.

```python
app.route_handler_method_map['/test'][method]
```

Reply:
Should the handler be of type `RouteHandlerType`, as the type in [`BaseScope`](https://github.com/starlite-api/starlite/blob/d6313981c9129da2f88efb09a9d14424733c781b/starlite/types/asgi_types.py#L78)?

Reply:
yup